client/core/src/com/cows/game/roundSimulation/RoundSimulator.kt

```diff
@@ -0,0 +1,115 @@
+package com.cows.game.roundSimulation
+
+import com.cows.game.map.Map
+import com.cows.game.roundSimulation.rawJsonData.*
+import com.cows.game.roundSimulation.simulationModels.TowerSimulationModel
+import com.cows.game.roundSimulation.simulationModels.UnitSimulationModel
+import kotlin.math.roundToInt
+
+class RoundSimulator {
+
+    //Input: attackInstruction, defendInstruction og GameState
+    //Output RoundSimulation
+    //TODO add gameState, to simulate money, store map etc.
+    fun simulate(defendInstruction : List<JsonTower>, attackInstruction : List<JsonUnit>) : JsonRoundSimulation {
+        val eventLog = mutableListOf<JsonTick>();
+
+        val path = Map.getPathCoordinates()
+
+        //init simulation
+        val units = mutableListOf<UnitSimulationModel>()
+        val towers = mutableListOf<TowerSimulationModel>()
+
+
+        var timeToNextSpawn = 0
+        attackInstruction.forEach{
```

Comment:
Here you can use map instead

`val units = attackInstruction.mapIndexed { index, unit -> UnitSimulationModel(unit.id, 1, 0, 2, 0, index * 2) }`

Reply:
Good point! I will also do that for the towers:)